packages/driver/src/cypress/log.ts

```diff
@@ -624,13 +598,53 @@ class LogManager {
         }
       }
 
-      // set the log on the command
-      const current = state('current')
+      let command = log.get('command')
 
-      if (current) {
-        current.log(log)
+      if (!command) {
+        command = state('current')
+        log.set('command', command)
       }
 
+      if (command) {
+        // See commands/asserting.ts for a detailed explanation of commandLogId.
+        // Short version: If onBeforeLog sets the same commandLogId as an existing log on the current command,
+        // then we want to merge the current instance into the existing one, rather than display a new log to
+        // the user.
+        let commandLogId = log.get('commandLogId')
+
+        if (commandLogId != null) {
```

Comment:
This logic is specific to `commands/asserting.ts` and commands are responsible for managing their own logs. Theres no need to move this logic here. This also puts additional attributes on the log that don't seem necessary which are stored on on the runner and the command.